app/tests/conftest.py

```diff
@@ -25,19 +24,25 @@ def mock_tweepy_global():
         yield
 
 
-@fixture(autouse=True, scope="function")
-def setup_teardown():
-    print()
-    print("Setup")
-    connect(DB_NAME, host="mongomock://localhost")
-    yield
```

Comment:
**One question:** How did the tests able to use database? Where did u setup database?

Reply:
``` python
@pytest.fixture
def client():
    with TestClient(app) as c:
```
he use the one in `main.py` but with `is_mock=True` when run test :V. I think that's how magic work :V

Reply:
Turns out `mongoengine` has direct implement related to `mongomock` in its `connect` function. [Ref](https://docs.mongoengine.org/apireference.html#connecting). That's sneaky af bro :))

> is_mock – explicitly use mongomock for this connection (can also be done by using [`](https://docs.mongoengine.org/apireference.html#id1)mongomock: // ` as db host prefix)